overlord/snapstate/snapmgr.go

```diff
@@ -633,6 +634,34 @@ func (m *SnapManager) ensureUbuntuCoreTransition() error {
 	return nil
 }
 
+// GenerateCookies creates snap cookies for snaps that are missing them (may be the case for snaps installed
+// before the feature of running snapctl outside of hooks was introduced, leading to a warning
+// from snap-confine).
+func (m *SnapManager) GenerateCookies(st *state.State) error {
+	var snapNames map[string]*json.RawMessage
```

Comment:
Do we need to lock the state here? In case that this is caller's responsibility could we perhaps indicate that in the documentation?